src/Neo/SmartContract/KeyBuilder.cs

```diff
@@ -29,8 +30,22 @@ public class KeyBuilder
         /// <param name="prefix">The prefix of the key.</param>
         public KeyBuilder(int id, byte prefix)
         {
-            Add(id);
```

Comment:
@Jim8y The bug was here, it was stored in the OS format, but it was readed in `LittleEndian`, in our test is the same, but in a `BigEndian` machine, it will fault.

Reply:
Its still wrong. You need to check the machine. Use
```csharp
if (BitConverter.IsLittleEndian)
    Id = BinaryPrimitives.ReadInt32LittleEndian(cache);
else
    Id = BinaryPrimitives.ReadInt32BigEndian(cache);
```